rules/preprocessing.smk

```diff
@@ -49,7 +49,8 @@ rule phone_readable_datetime:
         time_segments_type = config["TIME_SEGMENTS"]["TYPE"],
         include_past_periodic_segments = config["TIME_SEGMENTS"]["INCLUDE_PAST_PERIODIC_SEGMENTS"]
     output:
-        "data/raw/{pid}/phone_{sensor}_with_datetime.csv"
+        sensor_with_datetime = "data/raw/{pid}/phone_{sensor}_with_datetime.csv",
+        flag_file = touch("data/raw/{pid}/phone_{sensor}_with_datetime.done")
```

Comment:
Why do we need this flag file?

Reply:
Since a file is processed (read, add date time, and write) in chunks, a file named **_with_datetime.csv** will be generated at the first chunk and kept adding new rows in the rest chunks. If it stops running (e.g. accidentally shutting down the server), we might have an uncompleted output file named **_with_datetime.csv**. We are not sure whether all chunks have already been processed or not. That's why I think we need a flag file.